src/mmw/js/src/draw/views.js

```diff
@@ -500,64 +425,136 @@ var AoIUploadView = Marionette.ItemView.extend({
     }
 });
 
-var SelectAreaView = Marionette.ItemView.extend({
-    $label: $('#boundary-label'),
+var DrawToolBaseView = Marionette.ItemView.extend({
+    template: drawToolTmpl,
 
     ui: {
-        items: '[data-tile-url]',
-        button: '#predefined-shape',
-        helptextIcon: 'a.help'
+        drawToolButton: '.draw-tool-button',
+        drawToolItemButton: '.draw-pane-list-item-button',
+        helptextIcon: 'a.help',
+        resetButton: '.reset-draw-button'
     },
 
     events: {
-        'click @ui.items': 'onItemClicked'
+        'click @ui.drawToolItemButton': 'onClickItem',
+        'click @ui.drawToolButton': 'openDrawTool',
+        'click @ui.resetButton': 'resetDrawingState'
     },
 
     modelEvents: {
-        'change': 'render',
-        'change:toolsEnabled': 'removeBoundaryLayer'
+        'change': 'render'
+    },
+
+    initialize: function(options) {
+        this.resetDrawingState = options.resetDrawingState;
+        var self = this;
+
+        $(document).on('mouseup', function(e) {
+            var isTargetOutside = $(e.target).parents('.dropdown-menu').length === 0;
+            if (isTargetOutside && self.model.get('openDrawTool') === self.id) {
+                self.model.closeDrawTool();
+            }
+        });
     },
 
     templateHelpers: function() {
+        var activeDrawTool = this.model.get('activeDrawTool'),
+            activeDrawToolItem = this.model.get('activeDrawToolItem'),
+            currentZoomLevel = App.getLeafletMap().getZoom(),
+            openDrawTool = this.model.get('openDrawTool'),
+            activeTitle = null,
+            activeDirections = null,
+            toolData = this.getToolData();
+
+        if (activeDrawTool === toolData.id) {
+            var activeItem = _.find(toolData.items, function(item) {
+                return item.id === activeDrawToolItem;
+            });
+            activeTitle = activeItem.title;
+            activeDirections = activeItem.directions;
+        }
+
         return {
-            toolsDisabled: !this.model.get('toolsEnabled'),
-            currentZoomLevel: App.getLeafletMap().getZoom()
+            toolData: toolData,
+            activeTitle: activeTitle,
+            activeDirections: activeDirections,
+            currentZoomLevel: currentZoomLevel,
+            openDrawTool: openDrawTool
         };
     },
 
-    initialize: function() {
+    onShow: function() {
+        this.activatePopovers();
+    },
+
+    onRender: function() {
+        this.activatePopovers();
+    },
+
+    openDrawTool: function() {
+        this.model.openDrawTool(this.id);
+    },
+
+    activatePopovers: function() {
+        this.ui.helptextIcon.popover({
+            placement: 'top',
+            trigger: 'focus'
+        });
+    }
+});
+
+var SelectBoundaryView = DrawToolBaseView.extend({
+    $label: $('#boundary-label'),
+
+    initialize: function(options) {
+        DrawToolBaseView.prototype.initialize.apply(this, [options]);
```

Comment:
I'll switch to using `.call`. I'm kind of reluctant to use `super` though since it could be confusing to mix ES6 syntax with old school JS, especially since other parts of the project are using the latter.